jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/buffer/DirectBufferStatisticsHandler.java

```diff
@@ -0,0 +1,93 @@
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.UNIT_BUFFERS;
+import static io.opentelemetry.contrib.jfr.metrics.internal.RecordedEventHandler.defaultMeter;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.contrib.jfr.metrics.internal.RecordedEventHandler;
+import java.time.Duration;
+import java.util.Optional;
+import jdk.jfr.consumer.RecordedEvent;
+
+public final class DirectBufferStatisticsHandler implements RecordedEventHandler {
+  private static final String METRIC_NAME_USAGE = "process.runtime.jvm.buffer.usage";
+  private static final String METRIC_NAME_LIMIT = "process.runtime.jvm.buffer.limit";
+  private static final String METRIC_NAME_COUNT = "process.runtime.jvm.buffer.count";
+  private static final String METRIC_DESCRIPTION_USAGE = "Measure of memory used by buffers";
+  private static final String METRIC_DESCRIPTION_LIMIT =
+      "Measure of total memory capacity of buffers";
+  private static final String METRIC_DESCRIPTION_COUNT = "Number of buffers in the pool";
+  private static final String COUNT = "count";
+  private static final String MAX_CAPACITY = "maxCapacity";
+  private static final String MEMORY_USED = "memoryUsed";
+
+  private static final String EVENT_NAME = "jdk.DirectBufferStatistics";
+  private static final Attributes ATTR = Attributes.of(ATTR_POOL, "direct");
+
+  private volatile long usage = 0;
+  private volatile long limit = 0;
+  private volatile long count = 0;
+
+  public DirectBufferStatisticsHandler() {
+    initializeMeter(defaultMeter());
+  }
+
+  @Override
+  public void accept(RecordedEvent ev) {
+    if (ev.hasField(COUNT)) {
+      count = ev.getLong(COUNT);
+    }
+    if (ev.hasField(MAX_CAPACITY)) {
+      limit = ev.getLong(MAX_CAPACITY);
+    }
+    if (ev.hasField(MEMORY_USED)) {
+      usage = ev.getLong(MEMORY_USED);
+    }
+  }
+
+  @Override
+  public String getEventName() {
+    return EVENT_NAME;
+  }
+
+  @Override
+  public void initializeMeter(Meter meter) {
+    meter
+        .upDownCounterBuilder(METRIC_NAME_USAGE)
+        .setDescription(METRIC_DESCRIPTION_USAGE)
+        .setUnit(BYTES)
+        .buildWithCallback(
+            measurement -> {
+              measurement.record(usage, ATTR);
+            });
+    meter
+        .upDownCounterBuilder(METRIC_NAME_LIMIT)
+        .setDescription(METRIC_DESCRIPTION_LIMIT)
+        .setUnit(BYTES)
+        .buildWithCallback(
+            measurement -> {
+              measurement.record(limit, ATTR);
+            });
+    meter
+        .upDownCounterBuilder(METRIC_NAME_COUNT)
+        .setDescription(METRIC_DESCRIPTION_COUNT)
+        .setUnit(UNIT_BUFFERS)
+        .buildWithCallback(
+            measurement -> {
+              measurement.record(count, ATTR);
+            });
+  }
+
+  @Override
+  public Optional<Duration> getPollingDuration() {
+    return Optional.of(Duration.ofSeconds(1));
```

Comment:
What's the significance of the polling duration? Why 1s vs. 5s, 10s, etc?

Reply:
The JFR streaming system flushes the in memory JFR buffers holding event data to the disk repo at 1s intervals, at which point they may be consumed by the handlers. So polling at 1s intervals should provide the most up to date data without adding unnecessary overhead.